soroban-sdk/src/token.rs

```diff
@@ -39,16 +39,16 @@ pub trait Interface {
     /// * `spender` - The address being authorized to spend the tokens held by
     ///   `from`.
     /// * `amount` - The tokens to be made available to `spender`.
-    /// * `expiration_ledger` - The ledger number where this allowance expires. Cannot
+    /// * `live_until_ledger` - The ledger number where this allowance expires. Cannot
     ///    be less than the current ledger number unless the amount is being set to 0.
-    ///    An expired entry (where expiration_ledger < the current ledger number)
+    ///    An expired entry (where live_until_ledger < the current ledger number)
     ///    should be treated as a 0 amount allowance.
     ///
     /// # Events
     ///
     /// Emits an event with topics `["approve", from: Address,
-    /// spender: Address], data = [amount: i128, expiration_ledger: u32]`
-    fn approve(env: Env, from: Address, spender: Address, amount: i128, expiration_ledger: u32);
+    /// spender: Address], data = [amount: i128, live_until_ledger: u32]`
+    fn approve(env: Env, from: Address, spender: Address, amount: i128, live_until_ledger: u32);
```

Comment:
Similar to signatures, IMO this should stay as `expiration_ledger`. Approval is not recoverable, it truly expires and its semantics has nothing to do with state archival.

Reply:
This is something I discussed with @SirTyson. The justification for live_until_ledger was to remove "expiration" from the user phasing interface, and that the docs will mention live_until_ledgers for both persistent and temp entries.

Reply:
We shouldn't remove `expiration` from the user interface here. This interface has nothing to do with the storage - it's a token interface, and the approval actually expires and it can't be extended.

Reply:
I don't know what the underlying implementation is here, but if this value will ever be the input to the `extend` function, I think it should be `live_until_ledger` or `ttl`. However if this is just a field of a struct, then I think `expiration` is more appropriate. Like if we have this logic in the implementation we should definitely use `expiration`:
```
if currentLedger() > data.expiration:
   return error;
```

If that logic doesn't exist and this value is only used for extension, I'd use `live_until_ledger`.

Reply:
Yeah just double checked the example token contract, this is an actual field in the struct, not just the TTL value, so the term expiration is appropriate here. From `token/src/allowance.rs`:
```
    let allowance = AllowanceValue {
        amount,
        expiration_ledger,
    };

```

Reply:
I would say that the implementation details don't matter - this is a public interface and its semantics doesn't need to collide with state archival semantics. The semantics is 'approval expires on ledger X' - it just happens to be implemented via state archival (partially). In theory someone could come up with another implementation implemented via e.g. persistent entries.